### PR TITLE
Move file handling to writers

### PR DESCRIPTION
**Pull request by pljones**
Now files are only created if the writer is enabled.

### CODE REVIEW

**Comment by jonjamcam** [comment]
@pljones I just tested this branch and it is working ok now. No empty folders anymore. And if you check both options, you'll get the jam recorded twice, but this is up to the user if this is the intention. :+1:

File: src/Common/recorder/ClipSortLogGenerator.h [hunk 1]
```diff
@@ -17,6 +17,14 @@ class ClipSortLogGenerator : public JamMetadataWriter
     {
         return QCoreApplication::translate("Recorder::ClipSortLogGenerator", "Generate Reaper 'clipsort.log' file");
     }
+    inline void setJamDir(QString newJamName, QString recordBasePath) override {
```
Comment (elieserdejesus) [inline src/Common/recorder/ClipSortLogGenerator.h]: @pljones , a have just one little code review: too much code for a inline function here?

Comment (pljones) [inline src/Common/recorder/ClipSortLogGenerator.h]: Whoops yes - I'd forgotten to make the subdir originally.  I'll move it.

**Comment by pljones** [comment]
@elieserdejesus, I've updated per comments.

**Comment by elieserdejesus** [comment]
thanks @pljones.

**Pull request merged**